### ISSUE TITLE
Prefetching on hover is broken

### ISSUE THREAD

**Issue by Timer** [vercel/next.js#18856]
# Bug report

## Describe the bug

Next.js is supposed to prefetch the next page, even when prefetching is disabled, when a `<Link>` is **hovered**. The automated prefetch disabling is only supposed to affect the in-viewport prefetch.

## To Reproduce

```tsx
<Link prefetch={false} ...>
```

## Expected behavior

Prefetch on hover!

**Comment by balazsorban44** [comment]
Actually have been struggling with this for a while at work. I thought I did something wrong. We have a sitelist-like page with more than 50 links to other pages and prefetching them freezes up the whole app for like 40 sec (because each page's getStaticProps needs an access token to get some data from a third party api and getting like 50 access tokens from an oauth server at the same time is seemingly slow. I cache the access token until it expires for subsequent updates, but the first hit just kills the experience for that unlucky user) 

I tried setting prefetch to false but the prefetching kept happening anyway when links came into the viewport. Happy to see this, and hopefully will solve our issue.
